src/Microsoft.ML.Experimental/DataLoadSave/Database/SqlDatabaseReader.cs

```diff
@@ -0,0 +1,23 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Data.Common;
+using System.Data.SqlClient;
+
+namespace Microsoft.ML.Data
+{
+    public sealed class SqlDatabaseReader : DatabaseReader
+    {
+        public SqlDatabaseReader(string connectionString, string commandText) : base(connectionString, commandText)
+        {
+        }
+
+        protected override DbConnection CreateConnection()
```

Comment:
In the simplest scenario, the user just needs to implement `CreateConnection`. The three steps can be overridden and customized as needed, however.